File: 07-Encordados-JPA-Hibernate-Java/src/main/java/utils/ApplicationProperties.java
package utils;

import java.io.IOException;
import java.util.Properties;
import java.util.logging.Level;
import java.util.logging.Logger;

public class ApplicationProperties {

    private final Properties properties;

    public ApplicationProperties() {
        properties = new Properties();
        try {
            properties.load(getClass().getClassLoader().getResourceAsStream("application.properties"));

        } catch (IOException ex) {
            System.err.println("IOException Ocurrido al leer el fichero de propiedades: " + ex.getMessage());
            Logger.getLogger(getClass().getName()).log(Level.ALL, "IOException Ocurrido al leer el fichero de propiedades: " + ex.getMessage());
        }
    }

    public String readProperty(String keyName) {
        // Logger.getLogger(getClass().getName()).log(Level.INFO, "Leyendo propiedad " + keyName);
        return properties.getProperty(keyName, "No existe esa clave en el fichero de propiedades");
    }

}

File: 08-Encordados-JPA-Hibernate-Kotlin/src/main/kotlin/utils/ApplicationProperties.kt
package utils

import mu.KotlinLogging
import java.io.IOException
import java.util.*

val logger = KotlinLogging.logger {}
class ApplicationProperties {
    private val properties: Properties = Properties()

    init {
        try {
            properties.load(javaClass.classLoader.getResourceAsStream("application.properties"))
        } catch (ex: IOException) {
            System.err.println("IOException Ocurrido al leer el fichero de propiedades: " + ex.message)
            logger.error { "IOException Ocurrido al leer el fichero de propiedades: " + ex.message }
        }
    }

    fun readProperty(keyName: String?): String {
        // Logger.getLogger(getClass().getName()).log(Level.INFO, "Leyendo propiedad " + keyName);
        return properties.getProperty(keyName, "No existe esa clave en el fichero de propiedades")
    }
}